sui-execution/latest/sui-adapter/src/programmable_transactions/context.rs

```diff
@@ -872,7 +874,24 @@ mod checked {
                 }
             }
 
-            for (id, (recipient, ty, value)) in writes {
+            for (id, (recipient, tag, value)) in writes {
+                let ty = unwrap_type_tag_load(
+                    protocol_config,
+                    load_type_from_struct(
```

Comment:
can we just change the signature of `load_type_from_struct` to ensure we always use the type tag?

Reply:
Sorry but I'm not following I think :'( currently `load_type_from_struct` requires a `StructTag` which makes sense to me, however the `write`s hold a `MoveObjectType` which is a bit stronger of a guarantee (but any `MoveObjectType` is convertible to a `StructTag` which is what we're doing here).

Reply:
Ah, this is because this was to only place in places that had altered behavior from the previous protocol version. `load_type_from_struct` is called in a number of different places that remain unchanged in this PR, and changing the behavior in all of those places (while fine) seems a bit excessive IMHO.